ttnn/cpp/ttnn/operations/data_movement/pad/pad_pybind.hpp

```diff
@@ -0,0 +1,58 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include "ttnn/cpp/pybind11/decorators.hpp"
+
+#include "pad.hpp"
+
+namespace ttnn::operations::data_movement::detail {
+namespace py = pybind11;
+
+void bind_pad(py::module& module) {
+    auto doc =
+        R"doc(pad(input_tensor: ttnn.Tensor, padding: Tuple[Tuple[int, int], ...], value: Union[int, float], *, Optional[ttnn.MemoryConfig] = None) -> ttnn.Tensor
+Pad tensor with constant value. Padded shape is accumulated if ttnn.pad is called on a tensor with padding.
+
+Args:
+    * :attr:`input_tensor`: input tensor
+    * :attr:`padding`: padding to apply. Each element of padding should be a tuple of 2 integers, with the first integer specifying the number of values to add before the tensor and the second integer specifying the number of values to add after the tensor.
+    * :attr:`value`: value to pad with
+
+Keyword Args:
+    * :attr:`memory_config`: the memory configuration to use for the operation
+    * :attr:`queue_id` (Optional[uint8]): command queue id
+    * :attr:`use_multicore` (Optional[bool]): whether or not we should use multicore. Defaults to true
+    )doc";
+
+    using OperationType = decltype(ttnn::pad);
+    ttnn::bind_registered_operation(
+        module,
+        ttnn::pad,
+        doc,
+        ttnn::pybind_overload_t{
+            [] (const OperationType& self,
+                const ttnn::Tensor& input_tensor,
+                std::vector<std::pair<uint32_t, uint32_t>> padding,
+                const float value,
+                const std::optional<ttnn::MemoryConfig>& memory_config,
+                const bool use_multicore,
+                uint8_t queue_id) {
+                    return self(queue_id, input_tensor, padding, value, use_multicore, memory_config);
+                },
+                py::arg("input_tensor").noconvert(),
+                py::arg("padding"),
+                py::arg("value"),
+                py::kw_only(),
+                py::arg("memory_config") = std::nullopt,
+                py::arg("queue_id") = 0,
+                py::arg("use_multicore") = true
```

Comment:
Do we still have cases where `use_multicore` wants to be `False`? If not, I think we should get rid of it. This flag shouldn't have been exposed to users in the first place.

Reply:
oh true, I don't think we have a situation where its ever not true. I'll check

Reply:
not used anywhere (I do see that flag being used for untilize, future problem) . I can remove it here.

Reply:
yay, lets remove?